### PR TITLE
Add kernel volk_32fc_accumulator_s32fc

### PR DESCRIPTION
**Pull request by Ka-zam**
New PR. Fixed some bugs in stdddev_and_mean that somehow slipped through as well.

### CODE REVIEW

**Comment by jdemel** [comment]
This PR is a resubmission of #318 . There, we already have 2 approvals.
Though, this PR includes some fixes for stddev_and_mean as well.

File: kernels/volk/volk_32f_stddev_and_mean_32f_x2.h
Comment (jdemel) [inline kernels/volk/volk_32f_stddev_and_mean_32f_x2.h:179]: Why does the alignment need to be twice as much as expected? You doubled the other values as well.

**Review by jdemel — COMMENTED** [review]
I have one remark. But besides, this PR LGTM.

File: kernels/volk/volk_32f_stddev_and_mean_32f_x2.h
Comment (Ka-zam) [reply kernels/volk/volk_32f_stddev_and_mean_32f_x2.h:179]: Because of temporary confusion on my part. But I fixed it and added some alignment directives for NEON as well. I think this is correct now.

Comment (jdemel) [reply kernels/volk/volk_32f_stddev_and_mean_32f_x2.h:179]: I'd assume we need 16byte alignment for NEON according to the [docs](https://developer.arm.com/documentation/ddi0344/f/unaligned-data-and-mixed-endian-data-support/unaligned-data-access-support/neon-data-alignment).

**Comment by jdemel** [comment]
We have some random CI failures. I assume, we need to investigate those. `sum_of_poly` and `stddev_` are the two that fail randomly. I hope there aren't any other.
The failing CI test here is due to this random behavior.

**Comment by Ka-zam** [comment]
I set the relative inaccuracy threshold to 1e-4 as opposed to the default 1e-2 without too much investigation. 
The new kernel is more accurate as it should be but perhaps the threshold is a bit too tight. I will make some test runs and see what I can find.

**Comment by Ka-zam** [comment]
The reason for random failures are AFAICT that the generic implementation may fail on long vectors as the accumulator overflows. So the fix would be to implement the dual accumulator algorithm for the generic case.

**Comment by jdemel** [comment]
> The reason for random failures are AFAICT that the generic implementation may fail on long vectors as the accumulator overflows. So the fix would be to implement the dual accumulator algorithm for the generic case.

So the generic implementation uses a different algorithm? For the sake of more reliable CI, it would be better to use the same algorithm in the generic impl.
BUT: I didn't see any `accumulator` failures. For now, I'd assume this kernel, that we should discuss here, seems to be fine. Unfortunately, other kernels, e.g. `sum_of_poly` and `stddev` seem to fail randomly. I suggest to move this discussion to a new thread.

**Comment by Ka-zam** [comment]
Agreed, this kernel appears to be fine. It does use the same algorithm.

The random failures of stddev I think can be solved by updating the generic algorithm so that it doesn't overflow.

**Review by jdemel — APPROVED** [review]
This code received approval in #318 . It still looks good to me. Thanks!

**Review by michaelld — APPROVED** [review]
LGTM still ; @Ka-zam would you please squash everything into a single commit? There are some changes that are reverted within the commits; it will be much cleaner to just have a single commit with all of the changes. Simple squash keeping the commit messages as is, is fine with me. Thx!

**Comment by michaelld** [comment]
@Ka-zam looks like the squash got messed up ... can you please try again?

**Comment by jdemel** [comment]
This PR has 2 approvals and could be merged. I'm a bit hesitant though. 
@Ka-zam would it be correct to merge this PR, then move on to #469 to improve upon this PR? In that case, I'd like to have both PRs ready to be merged and then merge them both.

**Comment by Ka-zam** [comment]
Yes, let's try to make this a bit more clean. I'll take a look as soon as #469 is merged.

**Pull request merged**
